lib/assert.js

```diff
@@ -549,30 +549,38 @@ function compareExceptionKey(actual, expected, key, message, keys, fn) {
   }
 }
 
-function expectedException(actual, expected, msg, fn) {
+function expectedException(actual, expected, message, fn) {
   if (typeof expected !== 'function') {
-    if (isRegExp(expected))
-      return expected.test(actual);
-    // assert.doesNotThrow does not accept objects.
-    if (arguments.length === 2) {
-      throw new ERR_INVALID_ARG_TYPE(
-        'expected', ['Function', 'RegExp'], expected
-      );
+    // Handle regular expressions.
+    if (isRegExp(expected)) {
+      const str = String(actual);
+      if (expected.test(str))
+        return;
+
+      throw new AssertionError({
+        actual,
+        expected,
+        message: message || `The regular expression ${inspect(expected)} did ` +
+                            `not match the input:\n\n${inspect(str)}\n`,
```

Comment:
I'm not sure but I think it should be the other way: The input did not match the regular expression.
As an argument, the `match` method is on `String`s prototype.

Reply:
I agree and I tried to find an appropriate wording for it as well but I ended up with this because I could not find a good brief description that allowed to name the RegExp first. Do you have a suggestion to reword this?

Reply:
The reason for preferring the RegExp first is that it is mostly short and on a single line. The input could be a long multi line string.

Reply:
@targos what do you think about: `The input did not match the regular expression /foobar/. Input:\n\${input}`?

// cc @Trott

Reply:
I just updated the code with my suggestion.